xscen/diagnostics.py

```diff
@@ -97,535 +110,256 @@ def _invert_unphysical_temperatures(
-    )
-    if op == "distance":
-        diff = abs(sim - scen)
-    elif op == "improvement":
-        diff = abs(sim) - abs(scen)
-    elif op == "difference":  # op == 'diff':
-        diff = sim - scen
+    if isinstance(properties, (str, Path)):
+        logger.debug("Loading properties module.")
+        module = load_xclim_module(properties)
+        properties = module.iter_indicators()
+    elif hasattr(properties, "iter_indicators"):
+        properties = properties.iter_indicators()
+
+    try:
+        N = len(properties)
+    except TypeError:
+        N = None
     else:
-        raise NotImplementedError(
-            "Operation not implemented. Try 'distance', 'improvement' or 'difference' instead."
+        logger.info(f"Computing {N} properties.")
+
+    # select periods for ds
+    if period is not None and "time" in ds:
+        ds = ds.sel({"time": slice(str(period[0]), str(period[1]))})
+        date_start = ds.time.values[0]
+        date_end = ds.time.values[-1]
+    # select periods for ref_measure
+    if (
+        dref_for_measure is not None
+        and period is not None
+        and "time" in dref_for_measure
+    ):
+        dref_for_measure = dref_for_measure.sel(
+            {"time": slice(str(period[0]), str(period[1]))}
         )
-    pc = diff.plot(
-        ax=axim,
-        robust=True,
-        cmap=cmimpr,
-        center=0,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
 
-    fig.suptitle(title, fontsize="x-large")
-    axsm.set_title(sim.name.replace("_", " ").capitalize())
-    axsc.set_title(scen.name.replace("_", " ").capitalize())
-    axim.set_title(op.capitalize())
-    fig.tight_layout()
-    fig.colorbar(
-        ps,
-        cax=plt.subplot(gs[1, 0]),
-        shrink=0.5,
-        label=sim.attrs.get("long_name", sim.name),
-        orientation="horizontal",
+    if unstack:
+        ds = unstack_fill_nan(ds)
+
+    unit_conversion = unit_conversion or {}
+    for var, unit in unit_conversion.items():
+        ds[var] = convert_units_to(ds[var], unit)
+
+    prop = xr.Dataset()  # dataset with all properties
+    meas = xr.Dataset()  # dataset with all measures
+    for i, ind in enumerate(properties, 1):
+        if isinstance(ind, tuple):
+            iden, ind = ind
+        else:
+            iden = ind.identifier
+        logger.info(f"{i} - Computing {iden}.")
+
+        # Make the call to xclim
+        prop[iden] = ind(ds=ds)
+
+        # calculate the measure if a reference dataset is given for the measure
+        if dref_for_measure and iden in dref_for_measure:
+            meas[iden] = ind.get_measure()(sim=prop[iden], ref=dref_for_measure[iden])
+            # create a merged long_name
+            prop_ln = prop[iden].attrs.pop("long_name", "").replace(".", "")
+            meas_ln = meas[iden].attrs.pop("long_name", "").lower()
+            meas_ln = meas_ln.replace("between the simulation and the reference", "")
```

Comment:
Penses-tu qu'on devrait l'enlever des `measures` tout court? Ça n'apporte en effet pas grand chose.

Reply:
ben ça garde le nom de la mesure qui a été utilisé parce que le nom de la variable reste la propriété. Ça peut être utile pour les titres de figures...

Reply:
Hum, pas certain de comprendre la réponse. Je parlais ici du `"between the simulation and the reference"`. Si ça n'ajoute rien à la métadonné, pas besoin de le mettre dans les mesures de base.

Reply:
ah j'avais mal compris, désolé. D'accord, je vais l'enlever de xclim.